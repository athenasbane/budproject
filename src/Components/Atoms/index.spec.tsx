import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { Title, Container } from './index';

describe('<Title />', () => {
    it.each`
        textSize
        ${'small'}
        ${'medium'}
        ${'large'}
    `('should have the correct styles when props are passed too it', ({ textSize }) => {
        const tree = renderer
            .create(
                <Title textAlign="left" textSize={textSize as 'small' | 'medium' | 'large'}>
                    Test
                </Title>,
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});

describe('<Container />', () => {
    it.each`
        direction   | marginBotton | marginLeft
        ${'row'}    | ${5}         | ${5}
        ${'column'} | ${10}        | ${10}
    `('should have the styles when props are passed to it', ({ direction, marginBottom, marginLeft }) => {
        const tree = renderer
            .create(
                <Container direction={direction} marginBottom={marginBottom} marginLeft={marginLeft}>
                    Test
                </Container>,
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
