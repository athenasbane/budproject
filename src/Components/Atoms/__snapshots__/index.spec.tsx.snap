// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should have the styles when props are passed to it 1`] = `
<div
  className=""
  direction="row"
>
  Test
</div>
`;

exports[`<Container /> should have the styles when props are passed to it 2`] = `
<div
  className=""
  direction="column"
>
  Test
</div>
`;

exports[`<Title /> should have the correct styles when props are passed too it 1`] = `
<h1
  className=""
>
  Test
</h1>
`;

exports[`<Title /> should have the correct styles when props are passed too it 2`] = `
<h1
  className=""
>
  Test
</h1>
`;

exports[`<Title /> should have the correct styles when props are passed too it 3`] = `
<h1
  className=""
>
  Test
</h1>
`;
